lighthouse-core/report/v2/renderer/category-renderer.js

```diff
@@ -116,6 +116,99 @@ class CategoryRenderer {
   }
 
   /**
+   * @param {!ReportRenderer.AuditJSON} audit
+   * @param {number} scale
+   * @return {!Element}
+   */
+  _renderPerfHintAudit(audit, scale) {
+    const extendedInfo = /** @type {!CategoryRenderer.PerfHintExtendedInfo}
+        */ (audit.result.extendedInfo);
+
+    const element = this._dom.createElement('details', [
+      'lh-perf-hint',
+      `lh-perf-hint--${Util.calculateRating(audit.score)}`,
+      'lh-expandable-details',
+    ].join(' '));
+
+    const summary = this._dom.createChildOf(element, 'summary', 'lh-perf-hint__summary ' +
+        'lh-expandable-details__summary');
+    const titleEl = this._dom.createChildOf(summary, 'div', 'lh-perf-hint__title');
+    titleEl.textContent = audit.result.description;
+
+    const sparklineContainerEl = this._dom.createChildOf(summary, 'div', 'lh-perf-hint__sparkline');
+    const sparklineEl = this._dom.createChildOf(sparklineContainerEl, 'div', 'lh-sparkline');
+    const sparklineBarEl = this._dom.createChildOf(sparklineEl, 'div', 'lh-sparkline__bar');
+    sparklineBarEl.style.width = audit.result.rawValue / scale * 100 + '%';
+
+    const statsEl = this._dom.createChildOf(summary, 'div', 'lh-perf-hint__stats');
+    const statsMsEl = this._dom.createChildOf(statsEl, 'div', 'lh-perf-hint__primary-stat');
+    statsMsEl.textContent = audit.result.rawValue.toLocaleString() + ' ms';
+
+    this._dom.createChildOf(summary, 'div', 'lh-toggle-arrow', {title: 'See resources'});
+
+    if (extendedInfo.value.wastedKb) {
+      const statsKbEl = this._dom.createChildOf(statsEl, 'div', 'lh-perf-hint__secondary-stat');
+      statsKbEl.textContent = extendedInfo.value.wastedKb.toLocaleString() + ' KB';
+    }
+
+    const descriptionEl = this._dom.createChildOf(element, 'div', 'lh-perf-hint__description');
+    descriptionEl.appendChild(this._dom.createSpanFromMarkdown(audit.result.helpText));
+
+    if (audit.result.details) {
+      element.appendChild(this._detailsRenderer.render(audit.result.details));
+    }
+
+    return element;
+  }
+
+  /**
+   * @param {!Array<!ReportRenderer.AuditJSON>} audits
+   * @param {!ReportRenderer.GroupJSON} group
+   * @param {function(!ReportRenderer.AuditJSON): !Element=} renderAudit
+   * @return {!Element}
+   */
+  _renderAuditGroup(audits, group, renderAudit) {
```

Comment:
this is kind of an oddball, injecting a renderAudit function when everything so far doesn't do this. Rather than invert control, could this function just return the `auditGroupElem` that the caller will `appendChild` its audits to?